### PR TITLE
github+travis: split CI tasks between Travis and GitHub

### PR DESCRIPTION
**Pull request by guggero**
To free up build in Travis, we decided to run the non-flaky parts of
the CI pipeline in GitHub Workflows/Actions only. The integration tests
on the other hand are removed from GitHub because individual actions
cannot be restarted there which caused us to restart the whole workflow
if one test was flaky.

This split should give us the best of both worlds: Fast run of small
checks, linting and unit tests with an easy overview of what failed in
the PR directly. And more free build slots on Travis to do more advanced
integration tests on other architectures and/or operating systems. And
the option to restart a single flaky integration test on Travis.

@Roasbeef this means we should require all GitHub Workflows to pass for a PR to be merged once this is in.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,37 +26,22 @@ env:
...
   include:
-    - stage: Build
-      script:
-        - make rpc-check
-        - make unit pkg=... case=_NONE_
-        - make lint workers=1
-        - make btcd
-        - make release sys=windows-amd64
-        - make mobile-rpc
-        - go build --tags="mobile" ./mobile
```
Comment (halseth) [inline .travis.yml:48]: I think this check needs to be moved to actions now?

Comment (guggero) [reply .travis.yml:48]: Ah yes, I forgot. Added them to the GitHub workflow/actions.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -255,63 +255,3 @@ jobs:
         with:
           path-to-profile: coverage.txt
           parallel: true
-
```
Comment (bhandras) [inline .github/workflows/main.yml:214]: Not in the diff part but the GH workflow still runs `install_bitcoind.sh`, is that still needed?

Comment (guggero) [reply .github/workflows/main.yml:214]: Good eye for details, but unfortunately `bitcoind` is required for some of the unit tests.

Comment (bhandras) [reply .github/workflows/main.yml:214]: Good to know, thanks :)

**Comment by Roasbeef** [comment]
> @Roasbeef this means we should require all GitHub Workflows to pass for a PR to be merged once this is in.

Yezzir, activating!

**Pull request merged**